main.cpp

```diff
@@ -31,14 +31,27 @@
 #define MBED_CONF_APP_ESP8266_RX MBED_CONF_APP_WIFI_RX
 #include "easy-connect/easy-connect.h"
 
+#ifdef TARGET_STM
```

Comment:
There are better ways. You just need to use configuration mechanism from mbed OS.

For example: in .cpp code.
```
DigitalOut signal_led(MBED_CONF_APP_SIGAL_LED);
```

And then in `mbed_app.json`
```
{
    "config": {
        "SIGNAL_LED": "NC"
    },
    "target_overrides": {
        "NUCLEO_F429ZI": {
            "SIGNAL_LED": "LED1"
        }
    }
}
````

So on a undefined boards your digital pins would be in `NC` (Not Connected) state. I believe all boards implement the `NC` macro.

However, it is not a topic for this PR, but for a later refactoring work to be done.